### PR TITLE
Add explanation to the document of concat_examples

### PR DESCRIPTION
**Pull request by delta2323**
This PR proposes improvement of the document of `concat_examples`. It adds explanation on overall idea of what `concat_examples` does.

### CODE REVIEW

File: chainer/dataset/convert.py [hunk 1]
```diff
@@ -39,12 +39,15 @@ def to_device(device, x):
...
-    this function concatenates them along the newly-inserted first axis (called
-    `batch dimension`) into one array. The basic behavior is same for examples
-    consisting of multiple arrays, i.e., corresponding arrays of all examples
-    are concatenated.
-
+    This function converts "an arrays of tuples" into "a tuple of arrays".
+    Given an list of examples each of which consists of
+    "multiple" arrays, this function takes an array
+    in the same "relative" position within an example from each example,
+    concatenates them along the newly-inserted first axis
```
Comment (mitmul) [inline chainer/dataset/convert.py]: missing a conjunction?

**Comment by mitmul** [comment]
@delta2323 Thank you for this PR which improves the docstring. I think it's better to add a concrete example code after the explanation sentences. Could you work on it? (This task is actually marked as "TODO" by @beam2d here: https://github.com/chainer/chainer/pull/4164/files#diff-7a44933aa091044a62656b48203f3e74L63

File: chainer/dataset/convert.py [hunk 2]
```diff
@@ -39,12 +39,15 @@ def to_device(device, x):
...
     """Concatenates a list of examples into array(s).
 
-    Dataset iterator yields a list of examples. If each example is an array,
-    this function concatenates them along the newly-inserted first axis (called
-    `batch dimension`) into one array. The basic behavior is same for examples
-    consisting of multiple arrays, i.e., corresponding arrays of all examples
-    are concatenated.
-
+    This function converts "an arrays of tuples" into "a tuple of arrays".
+    Given an list of examples each of which consists of
```
Comment (Crissman) [inline chainer/dataset/convert.py]: Given a list of examples, each of which consists of
a list of elements, this function makes an array
by taking the element in the same position from each example
and concatenates them along the newly-inserted first axis
(called `batch dimension`) into one array.
It repeats this for all positions and returns the resulting arrays.

**Comment by delta2323** [comment]
I reflected comments and added examples.

**Comment by delta2323** [comment]
jenkins test this please.

**Comment by mitmul** [comment]
Great, LGTM

**Pull request merged**